kyaml/yaml/rnode.go

```diff
@@ -512,24 +526,57 @@ func (rn *RNode) SetAnnotations(m map[string]string) error {
 }
 
 // GetLabels gets the metadata labels field.
+// If specific labels are provided, only those are returned.
+// If no specific labels are provided, then all are returned.
 // If the field is missing, returns an empty map.
 // Use another method to check for missing metadata.
-func (rn *RNode) GetLabels() map[string]string {
+func (rn *RNode) GetLabels(labels ...string) map[string]string {
 	meta := rn.getMetaData()
 	if meta == nil {
 		return make(map[string]string)
 	}
-	return rn.getMapFromMeta(meta, LabelsField)
+	return rn.getMapFromMeta(meta, LabelsField, labels...)
 }
 
 // getMapFromMeta returns map, sometimes empty, from metadata.
-func (rn *RNode) getMapFromMeta(meta *RNode, fName string) map[string]string {
-	result := make(map[string]string)
-	if f := meta.Field(fName); !f.IsNilOrEmpty() {
-		_ = f.Value.VisitFields(func(node *MapNode) error {
-			result[GetValue(node.Key)] = GetValue(node.Value)
-			return nil
-		})
+func (rn *RNode) getMapFromMeta(meta *yaml.Node, fName string, fields ...string) map[string]string {
+	f := yNodeField(meta, fName)
+	if f == nil {
+		return make(map[string]string)
+	}
+	if f.Kind != MappingNode {
+		return nil
+	}
+	if len(fields) == 0 {
+		return fieldsAsStringMapAll(f)
+	}
+	return fieldsAsStringMapSome(f, fields...)
+}
+
+// fieldsAsStringMapAll returns a map of the node's content.
+func fieldsAsStringMapAll(n *yaml.Node) map[string]string {
```

Comment:
Is it possible to move some of these optimizations directly into `RNode.VisitFields`? `VisitFields` exists to iterate over the contents of a mapping node, so it feels redundant to reimplement and I think `VisitFields` is more readable.

Reply:
I'll give that a try and see how things perform.